### PR TITLE
Setup ESLint, Prettier with Husky and Commitlint

### LINKED ISSUE #12: Set Up Prettier and ESLint
## Description
Configure Prettier and ESLint to maintain code consistency and quality across the project.

https://dev.to/jsdevspace/setup-nextjs-14-project-with-eslint-prettier-tailwind-css-226j

## Motivation
To enforce a consistent coding style and catch potential errors early, we need to integrate Prettier and ESLint into our development workflow. This will help ensure that all code address to the same standards and improve overall code quality.

## Tasks
- [x] Install Prettier as development dependencies
- [x] Set up Prettier configuration file (.prettierrc)
- [x] Set up ESLint configuration file (.eslintrc.json)
- [x] Set up ignore files (.prettierignore / .eslintignore)
- [x] Integrate Prettier with ESLint
- [x] Configure ESLint rules based on project requirements
- [x] Configure and integrate pre-commit hooks with Husky, Commitlint and lint-staged
- [x] Document the Prettier and ESLint setup and usage guidelines in the project's README

### PR DESCRIPTION
**Pull request by whyleonardo**
## Description
This PR sets up ESLint, Prettier, Husky, and Commitlint to ensure consistent code quality and formatting across the project.

## Todo
- [x] Install Prettier as development dependencies
- [x] Set up Prettier configuration file (.prettierrc)
- [x] Set up ESLint configuration file (.eslintrc.json)
- [x] Set up ignore files (.prettierignore / .eslintignore)
- [x] Integrate Prettier with ESLint
- [x] Configure ESLint rules based on project requirements
- [x] Configure and integrate pre-commit hooks with Husky and Commitlint
- [x] Document the Prettier and ESLint usage guidelines in the project's README

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*.config.*
+node_modules
+.next
+.husky
```
Comment (RO-HSA) [inline .eslintignore:4]: Adiciona "dist" e "build"

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.prettierignore
+.gitignore
+.github
+.next
+.node_modules
+pnpm-lock.yaml
```
Comment (RO-HSA) [inline .prettierignore]: Adiciona "dist" e "build"

**Review by RO-HSA — CHANGES_REQUESTED** [review]
Faz as adições nos ignores

**Review by RO-HSA — APPROVED** [review]
lgtm

**Review by RO-HSA — APPROVED** [review]
LGTM

**Review by roberto-juniorjp — APPROVED** [review]
lgtm

**Pull request merged**